app/uploaders/article_image_uploader.rb

```diff
@@ -0,0 +1,53 @@
+class ArticleImageUploader < CarrierWave::Uploader::Base
+  include CarrierWave::MiniMagick
+
+  def store_dir
+    # This will be the directory path in the CMS
+    "uploads/#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
+  end
+
+  def extension_whitelist
+    %w[jpg jpeg gif png]
+  end
+
+  def filename
+    # don't try changing the name if original_filename is missing
+    return unless original_filename
+
+    # This will be the filename for all the processed/uploaded images
+    model.title.downcase.tr(' ', '_') + File.extname(original_filename)
```

Comment:
I am sure there is a way.

commit de53f565e6874286689454f69598b606b5d2760e added a way for the header to be named 'header-X' where X is (1..∞) based on how many articles are found published already on a given day.
    
**BUT** my concern is that this current strategy does not make sure the header is matched to the right article.
    
for example:
    
1. create 2 articles, article-1 and article-2, to be published on the same day, in that order
2. add a header to article-2 before you add one to article-1
3. now article-1 will map to header-2 and article-2 will map to header-1
    
That is a bit confusing (even typing it was confusing).

Let me think a bit more on how to make sure the order stays correct (without adding *multiple* SQL hits)